modules/kubernetes/aks-core/modules.tf

```diff
@@ -308,3 +308,24 @@ module "azad_kube_proxy" {
     scope         = var.azad_kube_proxy_config.k8dash_config.scope
   }
 }
+
+# Prometheus
+module "prometheus" {
+  depends_on = [kubernetes_namespace.tenant, module.opa_gatekeeper]
+
+  for_each = {
+    for s in ["prometheus"] :
+    s => s
+    if var.prometheus_enabled
+  }
+
+  source = "../../kubernetes/prometheus"
+
+  remote_write_enabled = var.prometheus_config.remote_write_enabled
+  remote_write_url     = var.prometheus_config.remote_write_url
+  remote_write_name    = var.prometheus_config.remote_write_name
+
+  volume_claim_enabled            = var.prometheus_config.volume_claim_enabled
```

Comment:
Check up on how AKS handles disks and AZs. Will Prometheus be bound to a specific AZ when using a PV? Do we need to use a specific storage?

Reply:
When you write specific AZ do you mean a specific azure zone?

Specifically about zones: https://docs.microsoft.com/en-us/azure/aks/availability-zones#azure-disks-limitations
```
Volumes that use Azure managed disks are currently not zone-redundant resources. Volumes cannot be attached across zones and must be co-located in the same zone as a given node hosting the target pod.

Kubernetes is aware of Azure availability zones since version 1.12. You can deploy a PersistentVolumeClaim object referencing an Azure Managed Disk in a multi-zone AKS cluster and Kubernetes will take care of scheduling any pod that claims this PVC in the correct availability zone.
```

So my understanding is that node 1 is scheduled in zone A and node 2 is scheduled in zone B the disks will be created in those zones. We just have to write a anti affinity rule that says that we shoulden't shedule prometheus in the same zones.
We should be able to use label: topology.kubernetes.io/zone or failure-domain.beta.kubernetes.io/zone.
Here I need someones AKS expertiese to tell me witch one is the best.

PVC info: https://docs.microsoft.com/en-us/azure/aks/concepts-storage
```
managed-csi - Uses Azure StandardSSD locally redundant storage (LRS) to create a Managed Disk. The reclaim policy ensures that the underlying Azure Disk is deleted when the persistent volume that used it is deleted. The storage class also configures the persistent volumes to be expandable, you just need to edit the persistent volume claim with the new size.
```

And from: https://docs.microsoft.com/en-us/azure/aks/azure-disks-dynamic-pv
```
Each AKS cluster includes four pre-created storage classes, two of them configured to work with Azure disks:

The default storage class provisions a standard SSD Azure disk.
Standard storage is backed by Standard SSDs and delivers cost-effective storage while still delivering reliable performance.
The managed-premium storage class provisions a premium Azure disk.
Premium disks are backed by SSD-based high-performance, low-latency disk. Perfect for VMs running production workload. If the AKS nodes in your cluster use premium storage, select the managed-premium class.

```

This disk is only for temporary storage, it will send all the data to Thanos. But if our pods restart we shoulden't lose the data like we would with a emptyDir.
But we also don't need to think about backups or anything like that, if we losse the data we should max loose an hour of data.

If we get higher demands from our customers we have the possability to use other storageClasses.

Reply:
Found docs about failure-domain.beta.kubernetes.io/zone and it's deprecated: https://kubernetes.io/docs/reference/labels-annotations-taints/#failure-domainbetakubernetesiozone

We should use topology.kubernetes.io/zone

Reply:
Created anti-affinity rule, see bellow in tagged comment.